### ISSUE TITLE
seq2seq lstm with large encoder,decoder vocabulary

### ISSUE THREAD

**Issue by akramahmadi66** [keras-team/keras#11884]
Dear all
I have used large vocabulary for encode and decode layer. In this model, I used fit_generator function to handle input of encode layer by smaller batches.
The problem is number of neuron for dense layer, since the number of output words is more than 500000. Can I use DNS layer output based on the power of two? For example, with 10 neurons produce 1024 number.  Is correct my idea? Does anyone have any other suggestions?

![seq2seq](https://user-images.githubusercontent.com/32474515/50085487-6a23b200-020f-11e9-8821-33544f4522e6.png)

**Comment by limapedro** [comment]
I thought the same, although I never tried to do it, there's a special function for this problem, because calculate the softmax for half a million classes is a huge bottleneck. The solution for your problem is to use log_softmax as an activation function, I would test your idea first, why not!

**Comment by msymp** [comment]
@akramahmadi66 , please let us know you have examined @limapedro suggestions and tested it. Thanks.

**Comment by ymodak** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!
